Mathlib/Data/List/Range.lean

```diff
@@ -42,33 +43,43 @@ theorem chain_range_succ (r : ℕ → ℕ → Prop) (n a : ℕ) :
   rw [range_succ_eq_map, chain_cons, and_congr_right_iff, ← chain'_range_succ, range_succ_eq_map]
   exact fun _ => Iff.rfl
 
+-- Verify that `finRange` is semireducible.
+example : finRange 3 = [0, 1, 2] := rfl
+
+theorem finRange_eq_pmap_range (n : ℕ) : finRange n = (range n).pmap Fin.mk (by simp) := by
+  apply List.ext_getElem <;> simp [finRange]
+
 @[simp]
-theorem finRange_zero : finRange 0 = [] :=
-  rfl
```

Comment:
This change stops this working as a `dsimp` lemma; does the `rfl` proof still work?

Reply:
Restored.